admin/class-primary-term-admin.php

```diff
@@ -8,12 +8,22 @@
 /**
  * Adds the UI to change the primary term for a post.
  */
-class WPSEO_Primary_Term_Admin {
+class WPSEO_Primary_Term_Admin implements WPSEO_WordPress_Integration {
 
 	/**
 	 * Constructor.
 	 */
-	public function __construct() {
+	public function register_hooks() {
+		global $pagenow;
+
+		if ( ! ( WPSEO_Metabox::is_post_overview( $pagenow ) || WPSEO_Metabox::is_post_edit( $pagenow ) ) ) {
+			return;
+		}
+
+		if ( ! current_user_can( 'manage_categories' ) ) {
```

Comment:
Can you explain this check in the technical choices in this pull?

Reply:
Well, I started this pull as a fix for https://github.com/Yoast/wordpress-seo/issues/12254, until I realized, I was working on the primary taxonomy selector and not the replacement variable. In the aforementioned issue the problem is that people who do not have the `manage_categories` capability are not allowed to retrieve a list of terms when using the `wp-core` data API. This is not the case in this PR so I will remove this check.